Qa python homework 3

bogol243 opened the pull request:
# Домашнее задание № 3

Прошу @snicks92 или @z1ll1 или @SK1995 проверить его.

Что было сделано:
Тесты API запускаются через флаг -m API
Написан клиент, с возможностью авторизации в API, и с функциями для управления сегментами
Написан тест на создание сегмента через API и проверку что сегмент создан
Написан тест на удаление сегмента через API
В UI тесте на удаление сегмента, сегмент для удаления теперь создаётся через API
В UI тесте на создание сегмента, сегмент после создания теперь удаляется через API

bogol243 committed:
Update .gitignore

bogol243 committed:
Update .gitignore

bogol243 committed:
Update requirements.txt

bogol243 committed:
Create browsers.json

bogol243 committed:
add homework files

bogol243 committed:
removed chrome_ver from settings and imports

bogol243 committed:
add homework 3

bogol243 committed:
rename file tests.py

bogol243 committed:
renamed file test.py

bogol243 committed:
conftest and pytest.ini moved to root

bogol243 committed:
homework 3

bogol243 committed:
Colect not only api-needed cookies

bogol243 committed:
Merge branch 'master' into QA-Python-homework-3

bogol243 committed:
Moved to root dir of project

bogol243 committed:
modified to use api

sniiick commented on tests_api/tests/test_api.py:
```diff
@@ -0,0 +1,41 @@
+    def test_delete(self, client: ApiClient, segment_id_for_delete):
+        seg_id = segment_id_for_delete
+
+        # Если сегмент удалился, вернётся True
+        # иначе вернётся Json с описанием ошибки
+        delete_res = client.delete_segment(seg_id)
```
А если АПИ отдало код 204, но по факту сегмент не удалился? Надо вызвать метод get_segment (можно под pytest.raises) чтобы в этом убедиться

sniiick reviewed (changes requested):
Все хорошо, даже 2 домашку поправил под АПИ. Мелкий коммент

bogol243 committed:
test_delete completed

test_delete now tries to get_segment() after deleting and ensures that deleted segment is not found

sniiick reviewed (approved):
Теперь хорошо
